xio-core/src/main/java/com/xjeffrose/xio/config/ConfigReloader.java

```diff
@@ -72,24 +87,60 @@ private Config parse(Reader reader) {
     return load(new File(file));
   }
 
-  private boolean performUpdate(Meta<T> oldValue, Meta<T> newValue) {
-    return !MessageDigest.isEqual(oldValue.digest, newValue.digest);
+  private boolean performUpdate(ConfigFileMetadata current, ConfigFileMetadata previous) {
+    return (current.lastModified > previous.lastModified
+        && !MessageDigest.isEqual(current.digest, previous.digest));
+  }
+
+  private ConfigFileMetadata loadMetaData(File configFile) throws IllegalStateException {
+    try {
+      MessageDigest digest = MessageDigest.getInstance("SHA-256");
+      DigestInputStream digester = new DigestInputStream(new FileInputStream(configFile), digest);
```

Comment:
i see a lot of `Closable` resources that are not being closed - these InputStreams and Readers should be closed ... consider using `try with resources`

Reply:
ok will do

Reply:
swapped to try with resources =p